pkg/controller/cluster/server/template.go

```diff
@@ -0,0 +1,15 @@
+package server
+
+var singleServerTemplate string = `
+if [ -d "{{.ETCD_DIR}}" ]; then
+	# if directory exists then it means its not an initial run
+	/bin/k3s server --cluster-reset --config  {{.INIT_CONFIG}} {{.EXTRA_ARGS}}
+fi
+/bin/k3s server --config {{.INIT_CONFIG}} {{.EXTRA_ARGS}}`
+
+var HAServerTemplate string = ` 
+if [ ${POD_NAME: -1} == 0 ] && [ ! -d "{{.ETCD_DIR}}" ]; then
+	/bin/k3s server --config {{.INIT_CONFIG}} {{.EXTRA_ARGS}}
+else 
+	/bin/k3s server --config {{.SERVER_CONFIG}} {{.EXTRA_ARGS}}
+fi`
```

Comment:
This is getting quite complex, since K3s can be configured in many ways (env vars, config files, commands, etc) I feel there is a better way.

Reply:
unfortunately this is because of raft problems with etcd, we are forced to handle different cases where the cluster is misconfigured